server/scsynth/SC_CoreAudio.h

```diff
@@ -268,8 +270,14 @@ class SC_CoreAudioDriver : public SC_AudioDriver {
 
     bool StopStart();
 
+    template <bool isClipping>
     void Run(const AudioBufferList* inInputData, AudioBufferList* outOutputData, int64 oscTime);
 
+    template <bool isClipping>
+    friend OSStatus appIOProc(AudioDeviceID device, const AudioTimeStamp* inNow, const AudioBufferList* inInputData,
+                              const AudioTimeStamp* inInputTime, AudioBufferList* outOutputData,
+                              const AudioTimeStamp* inOutputTime, void* defptr);
```

Comment:
why did you add this declaration to the header? it doesn't seem necessary, since appIOProc doesn't access any members in this class

Reply:
You're right! I remember I had to do it at some point, but looking again at it I can't see why.. I should have definitely put a comment there. I will try to remove it and figure this out, thanks!

Reply:
Ok, I've figured it out. That friend function was already there, and it's needed (if I understand correctly) for appIOProc to access private members of `*defptr`, such as `mInputBufList` here:
https://github.com/supercollider/supercollider/blob/76e595cda18a3fd760a38dd574bc93af92880ba0/server/scsynth/SC_CoreAudio.cpp#L1264

the only thing this PR added to the declaration is a `template <bool IsClipping>` statement.

Reply:
you should add the `template` part to the preexisting friend declaration rather than creating a new one

Reply:
ops, now I see it! Thanks for catching it, my bad